libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/weChatAdapter.ts

```diff
@@ -0,0 +1,373 @@
+/**
+ * @module wechat
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {Activity, ActivityTypes, BotAdapter, ConversationReference, ResourceResponse, TurnContext } from 'botbuilder-core';
+import { Storage } from 'botbuilder-core';
+import { WeChatClient } from './weChatClient';
+import { IRequestMessageBase, SecretInfo, IResponseMessageBase, ResponseMessageTypes, TextResponse, ImageResponse, NewsResponse, MusicResponse, MPNewsResponse, VideoResponse, VoiceResponse, UploadMediaResult, MessageMenuResponse } from './weChatSchema';
+import { WeChatMessageMapper } from './weChatMessageMapper';
+import * as xml2js from 'xml2js';
+import { VerificationHelper } from './verificationHelper';
+import { MessageCryptography } from './messageCryptography';
+
+/**
+ * Express or Restify Request object.
+ */
+export interface WebRequest {
+    body?: any;
+    headers: any;
+    on(event: string, ...args: any[]): any;
+}
+
+/**
+ * Express or Restify Response object.
+ */
+export interface WebResponse {
+    end(...args: any[]): any;
+    send(body: any): any;
+    status(status: number): any;
+}
+
+/**
+ * Settings used to configure a `WeChatAdapter` instance.
+ */
+export interface WeChatAdapterSettings {
+    AppId: string;
+    AppSecret: string;
+    Token: string;
+    EncodingAESKey: string;
+    UploadTemporaryMedia: boolean;
+}
+
+/**
+ * Represents a adapter that can connect a bot to WeChat endpoint.
+ */
+export class WeChatAdapter extends BotAdapter {
+    /**
+     * Key to get all response from bot in a single turn.
+     */
+    private TurnResponseKey = 'turnResponse';
```

Comment:
>'turnResponse' [](start = 30, length = 14)

since this is a key in turnstate you can use Symbol('turnResponse'). That guarantees your key won't collide with anyone else.